mycore-base/src/main/java/org/mycore/frontend/fileupload/MCRUploadHelper.java

```diff
@@ -75,14 +88,19 @@ public static void checkPathName(String path) throws MCRException {
         if (path.contains("../") || path.contains("..\\")) {
             throw new MCRException("File path " + path + " may not contain \"../\".");
         }
-        splitPath(path).forEach(pathElement -> {
+        List<String> pathPart = splitPath(path).collect(Collectors.toList());
+        pathPart.forEach(pathElement -> {
             checkNotEmpty(path, pathElement);
             checkOnlyDots(path, pathElement);
             checkTrimmed(pathElement);
             checkEndsWithDot(pathElement);
             checkReservedNames(pathElement);
             checkInvalidCharacters(pathElement);
         });
+        String actualFileName = pathPart.get(pathPart.size() - 1);
+        if(!FILE_NAME_PREDICATE.test(actualFileName)){
+            throw  new MCRException("File name does not match " + FILE_NAME_PATTERN + "!");
+        }
```

Comment:
We need a extra method parameter to disable this (filename-only) check, e.g. if we want to create a directory.